components/director/internal/domain/runtime/resolver.go

```diff
@@ -862,35 +863,46 @@ func (r *Resolver) EventingConfiguration(ctx context.Context, obj *graphql.Runti
 // deleteAssociatedScenarioAssignments ensures that scenario assignments which are responsible for creation of certain runtime labels are deleted,
 // if runtime doesn't have the scenarios label or is part of a scenario for which no scenario assignment exists => noop
 func (r *Resolver) deleteAssociatedScenarioAssignments(ctx context.Context, runtimeID string) error {
-	scenariosLbl, err := r.runtimeService.GetLabel(ctx, runtimeID, model.ScenariosKey)
-	notFound := apperrors.IsNotFoundError(err)
-	if err != nil && !notFound {
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
 		return err
 	}
 
-	if notFound {
-		return nil
-	}
-
-	scenarios, err := labelPkg.ValueToStringsSlice(scenariosLbl.Value)
+	tntMapping, err := r.tenantSvc.GetTenantByID(ctx, tnt)
 	if err != nil {
 		return err
 	}
 
-	for _, scenario := range scenarios {
-		scenarioAssignment, err := r.scenarioAssignmentService.GetForScenarioName(ctx, scenario)
+	for _, parentTenantID := range tntMapping.Parents {
+		ctxWithParentTenant := tenant.SaveToContext(ctx, parentTenantID, "")
+
+		scenariosLbl, err := r.runtimeService.GetLabel(ctxWithParentTenant, runtimeID, model.ScenariosKey)
```

Comment:
The `scenarioLabel` will soon be removed. Instead of relying on it we should list the ASAs where the `Tenant` property of the ASA is amoung the `tntMapping.Parents`. Then we can remove the assigned scenarios using the ASAs. This will reduce the complexity and will remove unneeded calls to the DB.